my-ansible-playbook.yml

```
@@ -0,0 +1,69 @@
+- name: set up to do app
+  hosts: webservers
+  remote_user: ec2-user
+  vars_prompt:
+    - name: trello_key
+      prompt: 'what is your trello key'
+      private: no
+      default: ''
+    - name: trello_token
+      prompt: 'what is your trello token'
+      private: no
+      default: ''
+
+  tasks:
+  - name: install git
+    ansible.builtin.yum:
+      
+      name: git
+      state: latest
+    become: yes
+ 
+  - name: install python
+    ansible.builtin.yum:
+      
+      name: python3
+      state: latest
+    become: yes 
+
+  - name: install poetry
+    ansible.builtin.shell: curl -sSL https://install.python-poetry.org | python3 -
+    args:
+      creates: /home/ec2-user/.local/bin/poetry 
+
+  - name: create directory
+    ansible.builtin.file:
+      path: /opt/todoapp
+      state: directory
+      owner: ec2-user
+    become: yes  
+
+  - name: Git checkout
+    ansible.builtin.git:
+      repo: 'https://github.com/Kelliep11/-DevOps-Course-Starter2.git'
+      dest: /opt/todoapp
+      version: main 
+
+  - name: install dependencies
+    ansible.builtin.shell: /home/ec2-user/.local/bin/poetry install
+    args:
+      chdir: /opt/todoapp
+
+  - name: Copy to Host
+    ansible.builtin.template:
+       src: .env.j2
+       dest: /opt/todoapp/.env
+
+  - name: copy todoapp.service
+    ansible.builtin.copy:
+      src: todoapp.service
+      dest: /etc/systemd/system/todoapp.service
+      mode: '777'
```

Mode '777' is giving all users read, write & execute permissions; that doesn't cause any particular problems here, but it is a bad habit to get into, particularly as it increases the attack surface for an attacker who's managed to get into your system. Instead, ideally we limit the permissions to the minimum required; so e.g. a nicer solution here can be to make the `ec2-user` the "owner" of this file, which means they have enough permissions but doesn't automatically grant everyone else the same permissions.
```suggestion
      owner: 'ec2-user'
```